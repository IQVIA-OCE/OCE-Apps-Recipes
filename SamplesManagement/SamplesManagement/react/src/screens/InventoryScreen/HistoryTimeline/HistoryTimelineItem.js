import React from 'react';
import { StyleSheet, View } from 'react-native';
import { Text, useTheme } from '@oce-apps/apollo-react-native';
import recordTypeIcons from './RecordTypeIcons';

import Icon from 'react-native-vector-icons/MaterialCommunityIcons';
import color from 'color';

const defaultTextColor = '#080707';

const HistoryTimelineItem = ({ item }) => {
  const icon = recordTypeIcons[item.recordType.DeveloperName];
  const theme = useTheme();
  const textColor = theme.dark ? color(defaultTextColor).negate().hex() : defaultTextColor;

  return (
    <View style={styles.container}>
      <View style={styles.leftSide}>
        <View style={styles.iconContainer}>
          <View style={{ ...styles.iconBg, backgroundColor: icon.color }}>
            <Icon name={icon.name} size={17} color="white" />
          </View>
          <View style={{ ...styles.line, backgroundColor: icon.color }} />
        </View>
        <View style={styles.detailsContainer}>
          <Text numberOfLines={1} style={[styles.recordNameText, { color: textColor }]}>
            {item.recordType.Name}
          </Text>
          <Text numberOfLines={2} style={[styles.productName, { color: textColor }]}>
            {item.productName}
          </Text>
          <Text style={styles.createdDate}>{item.transactionDateTime}</Text>
        </View>
      </View>
      <View style={styles.rightSide}>
        <Text style={styles.lotId}>{item.lotNumber}</Text>
        <View
          style={{ borderRadius: 50, backgroundColor: '#f88962', padding: 2 }}
        >
          <Text style={{ color: 'white', fontWeight: 'bold' }}>
            {item.quantity}
          </Text>
        </View>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  leftSide: {
    flexDirection: 'row',
    flexGrow: 1,
    flexBasis: 0,
  },
  iconContainer: {
    width: 24,
    marginRight: 12,
    alignItems: 'center',
  },
  iconBg: {
    padding: 3,
    borderRadius: 4,
    width: 24,
    height: 24,
    alignItems: 'center',
    alignContent: 'center',
    marginBottom: 2,
  },
  line: {
    flex: 1,
    width: 2,
    backgroundColor: 'grey',
  },
  recordNameText: {
    fontSize: 13,
    marginBottom: 5,
    fontWeight: '400',
  },
  productName: {
    fontSize: 13,
    marginBottom: 5,
    fontWeight: '400',
  },
  createdDate: {
    fontSize: 11,
    marginBottom: 24,
    fontWeight: '400',
    color: '#3e3e3c',
  },
  rightSide: {
    alignItems: 'flex-end',
    flexGrow: 1,
    flexBasis: 0,
  },
  lotId: {
    fontSize: 12,
    color: '#3e3e3c',
  },
});

export default HistoryTimelineItem;
