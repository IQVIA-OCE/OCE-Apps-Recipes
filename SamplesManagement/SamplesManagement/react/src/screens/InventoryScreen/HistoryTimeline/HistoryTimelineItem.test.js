import React from 'react';
import renderer from 'react-test-renderer';
import HistoryTimelineItem from './HistoryTimelineItem';
import { render } from '@testing-library/react-native';

const item = {
  recordType: {
    DeveloperName: 'AcknowledgementOfShipment',
    Name: '',
  },
  productName: '',
  transactionDateTime: '',
  lotNumber: '1337',
  quantity: '',
};

describe('HistoryTimelineItem', () => {
  it('should render properly', () => {
    const { getByText } = render(<HistoryTimelineItem item={item} />);

    expect(getByText(/1337/)).toBeTruthy();
  });
});
