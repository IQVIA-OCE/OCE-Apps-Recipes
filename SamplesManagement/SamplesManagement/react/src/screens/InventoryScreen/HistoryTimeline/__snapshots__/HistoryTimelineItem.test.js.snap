// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HistoryTimelineItem should render properly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
    }
  }
>
  <View
    style={
      Object {
        "flexBasis": 0,
        "flexDirection": "row",
        "flexGrow": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "marginRight": 12,
          "width": 24,
        }
      }
    >
      <View
        style={
          Object {
            "alignContent": "center",
            "alignItems": "center",
            "backgroundColor": "#ef6e64",
            "borderRadius": 4,
            "height": 24,
            "marginBottom": 2,
            "padding": 3,
            "width": 24,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 17,
              },
              undefined,
              Object {
                "fontFamily": "Material Design Icons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          Ôùï
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#ef6e64",
            "flex": 1,
            "width": 2,
          }
        }
      />
    </View>
    <View>
      <Text
        numberOfLines={1}
        style={
          Object {
            "color": "#080707",
            "fontSize": 13,
            "fontWeight": "400",
            "marginBottom": 5,
          }
        }
      >
        
      </Text>
      <Text
        numberOfLines={2}
        style={
          Object {
            "color": "#080707",
            "fontSize": 13,
            "fontWeight": "400",
            "marginBottom": 5,
          }
        }
      >
        
      </Text>
      <Text
        style={
          Object {
            "color": "#3e3e3c",
            "fontSize": 11,
            "fontWeight": "400",
            "marginBottom": 24,
          }
        }
      >
        
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flexBasis": 0,
        "flexGrow": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#3e3e3c",
          "fontSize": 12,
        }
      }
    >
      
    </Text>
    <View
      style={
        Object {
          "backgroundColor": "#f88962",
          "borderRadius": 50,
          "padding": 2,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontWeight": "bold",
          }
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
